dts/bindings/sensor/zephyr_thermistor.dtsi

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	/omit-if-no-ref/ thermistor_R25_10K_B_3974: thermistor-R25-10K-B-3974 {
+		status = "disabled";
+		compatible = "zephyr,thermistor-rt-table";
+		num-pairs = <16>;
+
+		/* Thermistor R/T Table for
+		 *   R25 = 10000 ohms
+		 *   B(25/85) = 3974
+		 */
+		sample-datum-0 {
+			temp-c = <(-25)>;
+			resistance = <130370>;
+			sample-index = <0>;
+		};
```

Comment:
This format seems unnecessarily verbose, how about converting temp-c and resistance to `array`? Then `DT_INST_PROP` would give you an array initializer, just validate that they are the same length with `DT_INST_PROP_LEN` and you'd get away without `num-pairs`, `sample-index` and all this child noise stuff.